File: client/python/mesh/system/mesh_scheduler.py
#
# Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
# TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
#
#
from typing import List

from mesh.kinds import Topic, Timeout
from mesh.macro import spi
from mesh.mpc import ServiceProxy
from mesh.prsim import Scheduler


@spi("mesh")
class MeshScheduler(Scheduler):

    def __init__(self):
        self.remote = ServiceProxy.default_proxy(Scheduler)

    def timeout(self, timeout: Timeout, duration: int) -> str:
        return self.remote.timeout(timeout, duration)

    def cron(self, cron: str, binding: Topic) -> str:
        return self.remote.cron(cron, binding)

    def period(self, duration: int, binding: Topic) -> str:
        return self.remote.period(duration, binding)

    def dump(self) -> List[str]:
        return self.remote.dump()

    def cancel(self, task_id: str) -> bool:
        return self.remote.cancel(task_id)

    def stop(self, task_id: str) -> bool:
        return self.remote.stop(task_id)

    def emit(self, topic: Topic) -> bool:
        return self.remote.emit(topic)

    def shutdown(self, duration: int) -> bool:
        return self.remote.shutdown(duration)

File: client/java/src/main/java/io/be/mesh/system/MeshScheduler.java
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */
package io.be.mesh.system;

import io.be.mesh.macro.SPI;
import io.be.mesh.mpc.ServiceProxy;
import io.be.mesh.prsim.Scheduler;
import io.be.mesh.struct.Timeout;
import io.be.mesh.struct.Topic;

import java.time.Duration;
import java.util.Set;

/**
 * @author coyzeng@gmail.com
 */
@SPI("mesh")
public class MeshScheduler implements Scheduler {

    private final Scheduler scheduler = ServiceProxy.proxy(Scheduler.class);

    @Override
    public String timeout(Timeout timeout, Duration duration) {
        return scheduler.timeout(timeout, duration);
    }

    @Override
    public String cron(String cron, Topic binding) {
        return scheduler.cron(cron, binding);
    }

    @Override
    public String period(Duration duration, Topic binding) {
        return scheduler.period(duration, binding);
    }

    @Override
    public Set<String> dump() {
        return scheduler.dump();
    }

    @Override
    public boolean cancel(String taskId) {
        return scheduler.cancel(taskId);
    }

    @Override
    public boolean stop(String taskId) {
        return scheduler.stop(taskId);
    }

    @Override
    public void emit(Topic topic) {
        scheduler.emit(topic);
    }

    @Override
    public boolean isShutdown() {
        return scheduler.isShutdown();
    }

    @Override
    public void shutdown(Duration duration) throws InterruptedException {
        //
    }
}
